services/dbaas/inmemorydb/replica_set.go

```diff
@@ -0,0 +1,417 @@
+			return utils.GenerateSetError(resourceName, "persistence_mode", err)
+		}
+	}
+
+	if replicaSet.Properties.EvictionPolicy != nil {
+		if err := d.Set("eviction_policy", *replicaSet.Properties.EvictionPolicy); err != nil {
+			return utils.GenerateSetError(resourceName, "eviction_policy", err)
+		}
+	}
+
+	if replicaSet.Properties.InitialSnapshotId != nil {
+		if err := d.Set("initial_snapshot_id", *replicaSet.Properties.InitialSnapshotId); err != nil {
+			return utils.GenerateSetError(resourceName, "initial_snapshot_id", err)
+		}
+	}
+
+	if replicaSet.Metadata.DnsName != nil {
+		if err := d.Set("dns_name", *replicaSet.Metadata.DnsName); err != nil {
+			return utils.GenerateSetError(resourceName, "dns_name", err)
+		}
+	}
+
+	if replicaSet.Properties.Resources != nil {
+		var resources []interface{}
+		resourceEntry := setResourceProperties(*replicaSet.Properties.Resources)
+		resources = append(resources, resourceEntry)
+		if err := d.Set("resources", resources); err != nil {
+			return utils.GenerateSetError(resourceName, "resources", err)
+		}
+	}
+
+	if replicaSet.Properties.Connections != nil {
+		var connections []interface{}
+		for _, connection := range *replicaSet.Properties.Connections {
+			connectionEntry := setConnectionProperties(connection)
+			connections = append(connections, connectionEntry)
+		}
+		if err := d.Set("connections", connections); err != nil {
+			return utils.GenerateSetError(resourceName, "connections", err)
+		}
+	}
+
+	if replicaSet.Properties.MaintenanceWindow != nil {
+		var maintenanceWindow []interface{}
+		maintenanceWindowEntry := setMaintenanceWindowProperties(*replicaSet.Properties.MaintenanceWindow)
+		maintenanceWindow = append(maintenanceWindow, maintenanceWindowEntry)
+		if err := d.Set("maintenance_window", maintenanceWindow); err != nil {
+			return utils.GenerateSetError(resourceName, "maintenance_window", err)
+		}
+	}
+
+	if replicaSet.Metadata != nil && replicaSet.Metadata.DnsName != nil {
+		if err := d.Set("dns_name", *replicaSet.Metadata.DnsName); err != nil {
+			return utils.GenerateSetError(resourceName, "dns_name", err)
+		}
+	}
+
+	return nil
+}
+
+func (c *InMemoryDBClient) SetSnapshotData(d *schema.ResourceData, snapshot inMemoryDB.SnapshotRead) error {
+	if snapshot.Id == nil {
+		return fmt.Errorf("expected a valid ID for InMemoryDB snapshot, but got 'nil' instead")
+	}
+	d.SetId(*snapshot.Id)
+	if snapshot.Metadata == nil {
+		return fmt.Errorf("response metadata should not be empty for InMemoryDB snapshot with ID: %v", *snapshot.Id)
+	}
+	var metadata []interface{}
+	metadataEntry := make(map[string]interface{})
+	if snapshot.Metadata.CreatedDate != nil {
+		metadataEntry["created_date"] = (*snapshot.Metadata.CreatedDate).Time.Format("2006-01-02T15:04:05Z")
+	}
+	if snapshot.Metadata.LastModifiedDate != nil {
+		metadataEntry["last_modified_date"] = (*snapshot.Metadata.LastModifiedDate).Time.Format("2006-01-02T15:04:05Z")
```

Comment:
same, try using `constant.DatetimeZLayout`

Reply:
@cbeti-ionos done in 1dcf9347fba8f716e31e7175b0c3df789054923a